Added Modrinth + Github Support

Leviaria opened the pull request:
# Add GitHub & Modrinth support

Adds support for downloading and updating plugins from GitHub releases and Modrinth, in addition to the existing SpigotMC support.

## What's new
- `get-github owner/repo` - Download from GitHub releases
- `get-modrinth project-id` - Download from Modrinth  
- `search-github` and `search-modrinth` commands
- Automatic platform detection when checking for updates

## How it works
When you run `check all` or `update all`, pluGET now tries:
1. SpigotMC (existing)
2. Modrinth (file hash matching - very accurate)
3. GitHub (name matching)

<img width="924" height="284" alt="image" src="https://github.com/user-attachments/assets/a92c18d0-7253-40fd-af72-e611d9b449c6" />

Don't be surprised—I used dev builds for some plugins that are unstable, so it upgraded/downgraded to the latest stable version. Everything works as it should. In addition, since WorldGuardExtraFlags was discontinued, it took the Reborn version, which is actively maintained and therefore smarter. Enjoy!

## Files added
- `src/platforms/github_handler.py`
- `src/platforms/modrinth_handler.py` 
- Updated existing files to integrate the new platforms

No breaking changes - everything works exactly like before, just with more plugin sources available.

Neocky commented on src/platforms/github_handler.py:
```diff
@@ -0,0 +1,268 @@
+        version = "latest"
+    
+    # Create download path similar to spiget implementation
+    download_path = get_download_path(config_values)
```
This will error when `sftp` or `ftp` is selected in the config and pluGET is trying to download a plugin from modrinth or GitHub.
<details>

<summary>Stacktrace of error</summary>

```console
pluGET >> get-modrinth OhduvhIc    
Traceback (most recent call last):
  File "\pluget.py", line 46, in <module>
    handle_input()
    ~~~~~~~~~~~~^^
  File "\src\handlers\handle_input.py", line 92, in handle_input
    download_modrinth_plugin(input_selected_object, featured_only)
    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "src\platforms\modrinth_handler.py", line 158, in download_modrinth_plugin
    _download_modrinth_file(download_url, download_plugin_path)
    ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\src\platforms\modrinth_handler.py", line 193, in _download_modrinth_file
    with open(download_path, 'wb') as f:
         ~~~~^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '\\plugins\\veinminer-paper-2.4.9.jar'
```

</details>

This needs to have the check for the connection method. Something like this from the `plugin_downloader.py` file will do:
```py
    if config_values.connection != "local":
        download_path = create_temp_plugin_folder()
    else:
        download_path = get_download_path(config_values)

```

Neocky commented on src/platforms/modrinth_handler.py:
```diff
@@ -0,0 +1,358 @@
+        version = "latest"
+    
+    # Create download path similar to other implementations
+    download_path = get_download_path(config_values)
```
```suggestion
    if config_values.connection != "local":
        download_path = create_temp_plugin_folder()
    else:
        download_path = get_download_path(config_values)

```

Neocky commented on src/platforms/modrinth_handler.py:
```diff
@@ -0,0 +1,358 @@
+    # check if plugin file is a proper .jar-file (same validation as existing code)
+    try:
+        with ZipFile(download_path, "r") as plugin_jar:
+            plugin_jar.open("plugin.yml", "r")
```
When testing with Veinminer from modrinth this will cause pluGET to say that the plugin was not correctly downloaded because the `plugin.yml` file was not found. Veinminer has only a `paper-plugin.yml` file in their root plugin archive. 
We test it the same way in the `plugin_downloader.py`. This needs to be fixed across pluGET so it will check if one of the files is present.  
I will create an issue for that. You can change it or leave it as this mirrors currently the same behaviour across all platforms.

Neocky reviewed (changes requested):
Thanks again for creating the PR 💜 
I have reviewed it and you just need to fix the `sftp` & `ftp` handling for modrinth and GitHub. See my more detailed comment. I already suggested code for it.
The last comment is only a note for myself. If you want to work on this you absolutely can or else I will fix that.

Neocky commented:
Oh and the Readme need to be updated for the new commands. I can do that when merging your PR when you don't have time for that. Just need your confirmation

Leviaria commented:
Hi @Neocky,

Thanks for the detailed feedback! I've implemented all the requested fixes:

**Changes made:**
- ✅ **SFTP/FTP handling**: Added proper connection checks in both `github_handler.py` and `modrinth_handler.py` - now uses `create_temp_plugin_folder()` for non-local connections
- ✅ **Plugin validation**: Extended validation to support both `plugin.yml` and `paper-plugin.yml` files (fixes compatibility issue)
- ✅ **Temp folder cleanup**: Added proper cleanup for SFTP/FTP downloads

The implementation now mirrors the existing `plugin_downloader.py` behavior exactly as you suggested.

Hope this looks good now! Let me know if you need any other adjustments.

Neocky reviewed (commented):
LGTM

Neocky commented:
Hey @Leviaria,

thanks for fixing the small issues. I merged your PR. 

Thanks again for your time and support! 💜
